File: CMS.Admin/src/models/BaiHat.ts


import { Album_BaiHat } from '@/models/Album_BaiHat';
import { TheLoai } from '@/models/TheLoai';
import { CaSy_BaiHat } from '@/models/CaSy_BaiHat'; 

export interface BaiHat { 
    BaiHatID: number;
    TenBaiHat: string;
    LinkFileNhac: string;
    AnhDaiDien: string;
    TieuDe: string;
    NgayDang: Date;
    TheLoaiID: number;
    HienThiTrangChu: boolean;
    Album_BaiHat: Album_BaiHat[];
    TheLoai: TheLoai;
    CaSy_BaiHat: CaSy_BaiHat[];
}

File: CMS.Models/Models/BaiHat.cs
namespace CMS.Models
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Data.Entity.Spatial;

    [Table("BaiHat")]
    public partial class BaiHat
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Required]
        public int BaiHatID { get; set; }

        [StringLength(50)]
        public string TenBaiHat { get; set; }

        public string LinkFileNhac { get; set; }

        public string AnhDaiDien { get; set; }

        [StringLength(50)]
        public string TieuDe { get; set; }

        public DateTime NgayDang { get; set; }

        public int? TheLoaiID { get; set; }

        public bool? HienThiTrangChu { get; set; }

        [InverseProperty("BaiHat")]
        public virtual ICollection<Album_BaiHat> Album_BaiHat { get; set; }

        [ForeignKey("TheLoaiID")]
        public virtual TheLoai TheLoai { get; set; }

        [InverseProperty("BaiHat")]
        public virtual ICollection<CaSy_BaiHat> CaSy_BaiHat { get; set; }
    }
}
